src/main/java/seedu/address/model/customer/Customer.java

```diff
@@ -85,6 +92,47 @@ public boolean isSameCustomer(Customer otherCustomer) {
             && otherCustomer.getName().equals(getName());
     }
 
+    /**
+     * Adds commission to current customer.
+     *
+     * @param commission new commission to add.
+     */
+    public void addCommission(Commission commission) {
+        requireAllNonNull(commission);
+        commissions.add(commission);
+    }
+
+    /**
+     * Returns true if a commission with the same identity as {@code commission} exists in the customer's
+     * commission list.
+     */
+    public boolean hasCommission(Commission commission) {
+        requireAllNonNull(commission);
+        return commissions.contains(commission);
+    }
+
+    /**
+     * Replaces the given person {@code target} in the list with {@code editedCommission}.
+     * {@code target} must exist in the address book.
+     * The commission identity of {@code editedCommission} must not be the same as another existing commission in the
+     * customer's commission list.
+     */
+    public void setCommission(Commission target, Commission editedCommission) {
+        assert hasCommission(target);
```

Comment:
Should this be an assert statement? I'm assuming you're trying to validate whether the target commission exists in the commissions but 

> An assertion failure indicates the programmer made a mistake in the code

and since they can be disabled, should not be used to do work. Rather, should an exception be thrown here instead?

Also is there a missing `requireAllNonNull(target, editedCommission)` check here? (Not that it's necessary since `setCommission` would do this check as well, but your other methods all do this check, so maybe it would be better to add this check as well for uniformity)?

Not very sure though, what do you and the rest think?

Reply:
Good point, I think all those checks within Customer are unnecessary since `UniqueCommissionList` actually does these checks and throws either `CommissionNotFoundException` or `DuplicateCommissionException`. The only reason I can see to leave them in is to avoid depending on UniqueCommissionList to check for those cases, but unless we plan to change the implementation of commissions, we can just remove all of those `assert` and `requireNonNull` checks out.